### PR TITLE
feat: add release for per-namespace

### PR DESCRIPTION
**Pull request by DariuszPorowski**
## Description

This pull request introduces a new, unified set of release and sync automation scripts for resource-type namespaces, establishing a single authoritative definition of what a namespace is and providing reusable Bash libraries for release and sync workflows. The changes add modular scripts for listing, bundling, and versioning namespaces, and ensure consistency between local and CI workflows.

The most important changes are:

**Shared Namespace Logic**

* Added `.github/scripts/lib-namespaces.sh`, a shared Bash library that defines what a resource-type namespace is, provides helpers for enumerating namespaces, and maps between category folders and namespace names. This script is sourced by both release and sync tooling to ensure consistency.

**Release Tooling**

* Added `.github/scripts/release/lib.sh`, a Bash library for per-namespace release scripts, providing helpers for version bumping and querying the latest version for a namespace, and sourcing the shared namespace logic.
* Added `.github/scripts/release/list-namespaces.sh`, a script that prints all releasable namespaces (e.g., `Radius.Data`), using the shared logic.
* Added `.github/scripts/release/build-namespace-bundle.sh`, a script to package all manifest YAMLs and adjacent READMEs for a namespace into a deterministic tarball, for use in releases.
* Added `.github/build/release.mk`, a Makefile with targets for listing namespaces, computing the next version, and building release bundles, mirroring the CI workflow steps for local use.

**Sync Automation**

* Added `.github/scripts/compute-radius-sync-payload.sh`, a script to compute the payload for repository dispatch events, determining which namespaces are affected by a push or release and generating a JSON payload for downstream automation.

Related GitHub Issue: 

## Contributor Checklist

- [ ] File names follow naming conventions and folder structure
- [ ] Platform engineer documentation is in README.md
- [ ] Developer documentation is the top-level description property
- [ ] Example of defining the Resource Type is in the developer documentation
- [ ] Example of using the Resource Type with a Container is in the developer documentation
- [ ] Verified the output of `rad resource-type show` is correct
- [ ] All properties in the Resource Type definition have clear descriptions
- [ ] Enum properties have values defined in `enum: []`
- [ ] Required properties are listed in `required: []` for every object property (not just the top-level properties)
- [ ] Properties about the deployed resource, such as connection strings, are defined as read-only properties and are marked as `readOnly: true`
- [ ] Recipes include a results output variable with all read-only properties set
- [ ] Environment-specific parameters, such as a vnet ID, are exposed for platform engineers to set in the Environment
- [ ] Recipes use the [Recipe context object](https://docs.radapp.io/reference/context-schema/) when possible
- [ ] Recipes are provided for at least one platform
- [ ] Recipes handle secrets securely
- [ ] Recipes are idempotent
- [ ] Resource types and recipes were tested

### CODE REVIEW

**Comment by brooke-hamilton** [comment]
**`.github/scripts/release/lib.sh` (lines 49–57, `rtc_latest_version`): `next-version.sh` aborts silently once a prerelease (or any non-`X.Y.Z`) tag exists for a namespace.**

```bash
git -C "$RTC_REPO_ROOT" tag --list "${ns}/v*" 2>/dev/null |
    while IFS= read -r tag; do
        version="${tag#"${ns}/v"}"
        [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] && echo "$version"
    done |
    sort -V | tail -n1
```

When the **last** tag iterated fails the stable-version regex, the final `[[ … ]] && echo` returns non-zero → the `while` returns non-zero → under `set -o pipefail` the whole pipeline fails. In `next-version.sh` (`set -euo pipefail`), `CURRENT="$(rtc_latest_version …)"` then trips `set -e` and the script exits `1` with no diagnostic (the stderr summary is never reached, so the "Resolve version" step just fails).

Because `git tag --list` orders `Radius.Data/v0.1.0-rc.1` *after* `Radius.Data/v0.1.0`, this fires on the workflow''s own supported flow. Reproduced in a throwaway repo:
- Tag `Radius.Data/v0.1.0-rc.1` present → resolve stable `v0.1.0` → **exit 1, empty `$GITHUB_OUTPUT`**.
- Also blocks cutting `rc.2` after `rc.1`, and any release once a trailing prerelease tag exists.

(Zero-tags and "last tag is stable" happen to pass — the loop body either never runs or ends on a matching `echo` — which is why the happy path looks fine.) Since the workflow exposes `prerelease_label`, this will bite the first time anyone cuts an RC.

Suggested fix — make the pipeline not carry a false exit status, e.g.:

```bash
rtc_latest_version() {
    local ns="$1"
    git -C "$RTC_REPO_ROOT" tag --list "${ns}/v*" 2>/dev/null |
        sed -n "s|^${ns}/v\([0-9]\+\.[0-9]\+\.[0-9]\+\)$|\1|p" |
        sort -V | tail -n1
}
```

or keep the loop but replace `[[ … ]] && echo` with an `if … then echo; fi` (an `if` with no matched branch returns 0).

File: .github/scripts/compute-radius-sync-payload.sh [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                # Only resource-type manifest YAML changes drive a re-sync.
+                case "$path" in
+                    *.yaml | *.yml) ;;
+                    *) continue ;;
+                esac
+                top="${path%%/*}"
+                if rtc_is_excluded_dir "$top"; then
+                    continue
+                fi
+                if rtc_is_category "$top"; then
```
Comment (kachawla) [inline .github/scripts/compute-radius-sync-payload.sh]: rtc_is_category only looks at the post-push checkout. If the last manifest in a namespace is deleted, or a namespace folder is renamed, the old namespace will not be included in the payload and Radius may retain stale manifests. Could we derive affected namespaces from both BEFORE_SHA and AFTER_SHA, including old paths for deletes/renames?

File: .github/workflows/release-namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+concurrency:
+  # One release at a time per namespace; never cancel an in-flight release.
+  group: release-namespace-${{ inputs.namespace }}
+  cancel-in-progress: false
+
+jobs:
+  release:
+    name: Release ${{ inputs.namespace }}
+    if: github.repository == 'radius-project/resource-types-contrib'
```
Comment (kachawla) [inline .github/workflows/release-namespace.yaml]: Should releases be restricted to main? workflow_dispatch allows the operator to select a branch, and this job currently publishes a release from whichever ref was selected.

Could we allow dry runs from any branch but require main for a real release?

if: >-
  github.repository == 'radius-project/resource-types-contrib' &&
  (inputs.dry_run || github.ref == 'refs/heads/main')

Else an accidental run from a feature branch could create an official namespace tag and trigger the downstream Radius sync before the change is merged.

File: .github/scripts/release/next-version.sh [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+CURRENT="$(rtc_latest_version "$NAMESPACE")"
+BASE="${CURRENT:-0.0.0}"
+NEXT="$(rtc_semver_bump "$BASE" "$BUMP")"
+
+IS_PRERELEASE="false"
+if [[ -n "$PRERELEASE_LABEL" ]]; then
+    # Normalize: allow the caller to pass either "rc.1" or "-rc.1".
+    PRERELEASE_LABEL="${PRERELEASE_LABEL#-}"
+    # Restrict to the SemVer prerelease charset so the resulting tag is valid.
+    if [[ ! "$PRERELEASE_LABEL" =~ ^[0-9A-Za-z]([0-9A-Za-z.-]*[0-9A-Za-z])?$ ]]; then
```
Comment (kachawla) [inline .github/scripts/release/next-version.sh]: Should we tighten this validation a bit? Labels like `rc..1` or `rc.01` may still pass even though they are not valid SemVer prerelease identifiers.

**Review by kachawla — COMMENTED** [review]
Should we add a few tests for the new release and sync scripts, especially namespace deletion/rename and prerelease versioning edge cases?

File: .github/scripts/release/next-version.sh (hunk 3 above)
Comment (DariuszPorowski) [reply .github/scripts/release/next-version.sh]: done :)

File: .github/workflows/release-namespace.yaml (hunk 2 above)
Comment (DariuszPorowski) [reply .github/workflows/release-namespace.yaml]: done :)

File: .github/scripts/compute-radius-sync-payload.sh (hunk 1 above)
Comment (DariuszPorowski) [reply .github/scripts/compute-radius-sync-payload.sh]: done :)

**Review by kachawla — APPROVED** [review]

**Review by sk593 — APPROVED** [review]
